chacractl/api/binaries.py

```diff
@@ -71,27 +86,44 @@ def post(self, url, filepath):
             return self.put(file_url, filepath)
         elif exists.status_code == 404:
             logger.info('POSTing file: %s', filepath)
-            with open(filepath, 'rb') as binary:
```

Comment:
this `with` was closing the file at the end, with these changes that wouldn't be the case anymore.

If `load_file` would only calculate the `digest` instead of doing both the digest and returning the open file this code could stay the same.